sys/evtimer/evtimer.c

```diff
@@ -71,29 +72,31 @@ static void _del_event_from_list(evtimer_t *evtimer, evtimer_event_t *event)
             list->next = event->next;
             if (list->next) {
                 list_entry = list->next;
-                list_entry->offset += event->offset;
+                list_entry->offset_ms += event->offset_ms;
             }
             break;
         }
         list = list->next;
     }
 }
 
-static void _set_timer(xtimer_t *timer, uint32_t offset)
+static void _set_timer(xtimer_t *timer, uint32_t offset_ms)
 {
-    uint64_t offset_in_us = (uint64_t)offset * 1000;
+    uint64_t offset_in_us = (uint64_t)offset_ms * 1000;
 
-    DEBUG("evtimer: now=%" PRIu32 " setting xtimer to %" PRIu32 ":%" PRIu32 "\n",
-          xtimer_now_usec(), (uint32_t)(offset_in_us >> 32),
-          (uint32_t)(offset_in_us));
-    _xtimer_set64(timer, offset_in_us, offset_in_us >> 32);
+    DEBUG("evtimer: now=%" PRIu32 " us setting xtimer to %" PRIu32 ":%" PRIu32 " us\n",
+          xtimer_now_usec(), (uint32_t)(offset_in_us >> 32), (uint32_t)(offset_in_us));
+
+    xtimer_set64(timer, offset_in_us);
```

Comment:
Yes, but the previous solution was wrong, since it assumed `ticks == us`.

Reply:
I think this should have been in #8858, but got lost by splitting.

Reply:
Agreed... :-/... I know this is a big favor to ask, because you already split your original PR quite up a bit, but facing the release: could you maybe split out this change also, so we can merge this ASAP?

Reply:
@miri64 see #8963.